nixos/modules/services/matrix/conduwuit.nix

```diff
@@ -0,0 +1,299 @@
+          `systemd.services.conduwuit.serviceConfig.DynamicUser` has to be disabled
+        '';
+      }
+    ];
+
+    # DynamicUser requires both user and group to exist or both not to exist.
+    # If we're configured to run under a different user or group, ensure
+    # the other default exists.
+    users =
+      lib.mkIf
+        (
+          cfg.settings.global.unix_socket_path != null || cfg.group != defaultGroup || cfg.user != defaultUser
+        )
+        {
+          groups = lib.mkIf (cfg.group == defaultGroup) {
+            ${defaultGroup} = { };
+          };
+          users = lib.mkIf (cfg.user == defaultUser) {
+            ${defaultUser} = {
+              group = cfg.group;
+              home = cfg.settings.global.database_path;
+              isSystemUser = true;
+            };
+          };
+        };
+
+    systemd.services.conduwuit = {
+      description = "Conduwuit Matrix Server";
+      documentation = [ "https://conduwuit.puppyirl.gay/" ];
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      environment = lib.mkMerge ([
+        { CONDUWUIT_CONFIG = configFile; }
+        cfg.extraEnvironment
+      ]);
+      startLimitBurst = 5;
+      startLimitIntervalSec = 60;
+      serviceConfig = {
+        DynamicUser = true;
+        User = cfg.user;
+        Group = cfg.group;
+
+        DevicePolicy = "closed";
+        LockPersonality = true;
+        MemoryDenyWriteExecute = true;
+        NoNewPrivileges = true;
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        PrivateDevices = true;
+        PrivateMounts = true;
+        PrivateTmp = true;
+        PrivateUsers = true;
+        PrivateIPC = true;
+        RemoveIPC = true;
+        RestrictAddressFamilies = [
+          "AF_INET"
+          "AF_INET6"
+          "AF_UNIX"
+        ];
+        RestrictNamespaces = true;
+        RestrictRealtime = true;
+        SystemCallArchitectures = "native";
+        SystemCallFilter = [
+          "@system-service"
+          "@resources"
+          "~@clock"
+          "@debug"
+          "@module"
+          "@mount"
+          "@reboot"
+          "@swap"
+          "@cpu-emulation"
+          "@obsolete"
+          "@timer"
+          "@chown"
+          "@setuid"
+          "@privileged"
+          "@keyring"
+          "@ipc"
+        ];
+        SystemCallErrorNumber = "EPERM";
+
+        StateDirectory = "conduwuit";
+        StateDirectoryMode = "0700";
+        RuntimeDirectory = "conduwuit";
+        RuntimeDirectoryMode = "0750";
+
+        ExecStart = lib.getExe cfg.package;
+        Restart = "on-failure";
+        RestartSec = 10;
+        TimeoutStopSec = "4m";
+        TimeoutStartSec = "4m";
```

Comment:
I am not convinced that we should set those extra timeouts

Reply:
The linked systemd unit also has no explanation why the default should be changed. In question this might just delay a shutdown/restart/rebuild unnecessarily in case of an error.

Reply:
Fair enough, I just assumed there was a particular reason for upstream doing it. I'll remove the increased timeouts.